internal/datastore/postgres/readwrite.go

```diff
@@ -725,7 +725,7 @@ func exactRelationshipDifferentCaveatClause(r *core.RelationTuple) sq.And {
 			colUsersetRelation:  r.Subject.Relation,
 		},
 		sq.Or{
-			sq.Expr(`? IS DISTINCT FROM ?`, colCaveatContextName, caveatName),
```

Comment:
Can you describe why this fixes the test issue? Both look equivalent to me

Reply:
> Can you describe why this fixes the test issue? Both look equivalent to me


In the first one (`? IS DISTINCT FROM ?`) the `caveat_name` parameter is interpreted as a string value, and so all the caveat names get checked against that, and they are always distinct from that (if you dont try to use a caveat named `caveat_name`). If it is directly written into the Expression (`caveat_name IS DISTINCT FROM ?`) it gets interpreted as the column name and the value of that is used to make the comparison.

So the difference is one will look like this in sql `"caveat_name" IS DISTINCT FROM "some_caveat_name"` and the other looks like `caveat_name IS DISTINCT FROM "some_caveat_name"` in the end.

Reply:
makes sense, thanks for clarifying!